### ISSUE TITLE
New Rule: Docker Personal Access Token (@secretlint/secretlint-rule-docker)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1442]
## Description

Add a new rule to detect Docker Personal Access Tokens (PATs).

Docker PATs use the `dckr_pat_` prefix, making them reliably detectable with low false positive rate.

## Source

- [Docker Access Tokens documentation](https://docs.docker.com/security/access-tokens/) — **does not officially document the `dckr_pat_` prefix or token format**
- [GitHub Secret Scanning Push Protection: supported](https://docs.github.com/en/code-security/secret-scanning/introduction/supported-secret-scanning-patterns) — **independently corroborated**: when drafting this issue, synthetic `dckr_pat_` + 27-character samples were rejected by GitHub Secret Scanning as *high confidence* Docker PATs, indicating that Docker is a validator-partner and the format is stable enough for GitHub to detect with high confidence.
- [TruffleHog dockerhub v2 detector](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/dockerhub/v2/dockerhub.go) — community-observed prefix, regex: `dckr_pat_[a-zA-Z0-9_-]{27}`

## Format

Prefix `dckr_pat_` followed by exactly 27 alphanumeric, underscore, or hyphen characters (based on community observation; TruffleHog uses the same fixed length).

Regex (with Unicode boundary and trailing boundary, ReDoS-safe):
```
/(?<!\p{L})dckr_pat_[A-Za-z0-9_-]{27}(?![A-Za-z0-9_-])/gu
```

Note: Docker also issues Organization Access Tokens (OAT) with the `dckr_oat_` prefix followed by 32 characters. OAT support can be added as a separate message in a future iteration.

## Analysis

- `dckr_pat_` is unique; FP risk is low.
- Observed PATs are a fixed 27 characters. The original range 20–60 was too wide and would match truncated or fabricated tokens.
- Added trailing boundary `(?![A-Za-z0-9_-])` to keep reported ranges precise.
- **External validation**: GitHub Secret Scanning classifies `dckr_pat_` + 27 chars as a *high confidence* Docker PAT. Synthetic samples authored for this issue were rejected at push time, which is independent evidence that the format is stable and FP risk is very low.

## OK (should NOT be detected)

```
# placeholder / too short
dckr_pat_abc
```

## NG (should be detected)

A Docker PAT has the shape `dckr_pat_` followed by exactly 27 characters from `[A-Za-z0-9_-]`:

```
DOCKER_TOKEN=dckr_pat_<27-character body>
token: "dckr_pat_<27-character body>"
```

(Full sample tokens are intentionally omitted here because GitHub Secret Scanning rejects matching literals as high-confidence Docker PATs. The rule's test fixtures should provide real-shape synthetic samples in a non-scanned location.)

**Comment by azu** [comment]
https://github.com/secretlint/secretlint/tree/master/packages/%40secretlint/secretlint-rule-docker
